ballista/rust/executor/src/cpu_bound_executor.rs

```diff
@@ -0,0 +1,376 @@
+        } else {
+            d.field("thread", &"None")
+        };
+
+        d.finish()
+    }
+}
+
+impl DedicatedExecutor {
+    /// https://stackoverflow.com/questions/62536566
+    /// Creates a new `DedicatedExecutor` with a dedicated tokio
+    /// runtime that is separate from the `[tokio::main]` threadpool.
+    ///
+    /// The worker thread priority is set to low so that such tasks do
+    /// not starve other more important tasks (such as answering health checks)
+    ///
+    pub fn new(thread_name: impl Into<String>, num_threads: usize) -> Self {
+        let thread_name = thread_name.into();
+        let name_copy = thread_name.to_string();
+
+        let (tx, rx) = std::sync::mpsc::channel();
+
+        //Cannot create a seperated tokio runtime in another tokio runtime,
+        //So use std::thread to spawn a thread
+        let thread = std::thread::spawn(move || {
+            let runtime = tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .thread_name(&name_copy)
+                .worker_threads(num_threads)
+                .on_thread_start(move || set_current_thread_priority(WORKER_PRIORITY))
+                .build()
+                .expect("Creating tokio runtime");
+
+            // By entering the context, all calls to `tokio::spawn` go
+            // to this executor
+            let _guard = runtime.enter();
+
+            while let Ok(request) = rx.recv() {
+                // TODO feedback request status
+                tokio::task::spawn(request);
+            }
+        });
+
+        let state = State {
+            num_threads,
+            thread_name,
+            requests: Some(tx),
+            thread: Some(thread),
+        };
+
+        Self {
+            state: Arc::new(Mutex::new(state)),
+        }
+    }
+
+    /// Runs the specified Future (and any tasks it spawns) on the
+    /// `DedicatedExecutor`.
+    ///
+    /// Currently all tasks are added to the tokio executor
+    /// immediately and compete for the threadpool's resources.
+    pub fn spawn<T>(&self, task: T) -> Receiver<T::Output>
+    where
+        T: Future + Send + 'static,
+        T::Output: Send + 'static,
+    {
+        let (tx, rx) = tokio::sync::oneshot::channel();
+
+        // create a execution plan to spawn
+        let job = Box::pin(async move {
+            let task_output = task.await;
+            if tx.send(task_output).is_err() {
+                warn!("Spawned task output ignored: receiver dropped");
+            }
+        });
+
+        let mut state = self.state.lock();
+
+        if let Some(requests) = &mut state.requests {
+            // would fail if someone has started shutdown
+            requests.send(job).ok();
+        } else {
+            warn!("tried to schedule task on an executor that was shutdown");
+        }
+
+        rx
+    }
+
+    /// signals shutdown of this executor and any Clones
+    #[allow(dead_code)]
```

Comment:
I wonder if this `#allow` is really necessary?

Reply:
For fix `clippy`, In the future if realized close the executor, will delete this.